feat: add AWS Bedrock AgentCore to tools registry and documentation

siwachabhi opened the pull request:
## Summary
- Add AWS Bedrock AgentCore to the tools registry with code execution and browser capabilities
- Add AWS Bedrock AgentCore to the tools documentation under Ready-to-Use Tool Packages

## Changes
1. **Tools Registry** (`content/tools-registry/registry.ts`)
   - Added bedrock-agentcore package entry with CodeInterpreterTools and BrowserTools
   - Includes OIDC authentication setup via Vercel AWS credentials provider
   - Tags: code-execution, browser-automation, sandbox, web-automation, aws, bedrock, agentcore

2. **Documentation** (`content/docs/02-foundations/04-tools.mdx`)
   - Added AWS Bedrock AgentCore to the Toolsets section
   - Links to GitHub repository and AWS documentation

## Links
- GitHub: https://github.com/aws/bedrock-agentcore-sdk-typescript
- AWS Docs: https://docs.aws.amazon.com/bedrock-agentcore/latest/devguide/built-in-tools.html
- NPM: https://www.npmjs.com/package/bedrock-agentcore

Commit:
feat(tools-registry): add AWS Bedrock AgentCore integration

Add bedrock-agentcore package with code execution and browser automation tools. Includes OIDC authentication setup via Vercel AWS credentials provider.

Commit:
docs(tools): add AWS Bedrock AgentCore to toolsets

Add AWS Bedrock AgentCore built-in tools to the Ready-to-Use Tool Packages section with links to GitHub repository and AWS documentation for code execution and browser capabilities.

Commit:
docs(tools): update AWS Bedrock AgentCore description

Use proper capitalization for Code Interpreter and Browser tools.

Commit:
feat(tools-registry): update AWS Bedrock AgentCore code example

Add try/finally block with stopSession() cleanup for proper resource management. Update model ID and prompt to demonstrate combined Browser and Code Interpreter usage.

kosti-v commented on content/docs/02-foundations/04-tools.mdx:
```diff
@@ -153,6 +153,7 @@ These packages provide pre-built tools you can install and use immediately:
 - **[Stripe agent tools](https://docs.stripe.com/agents?framework=vercel)** - Tools for interacting with Stripe.
 - **[StackOne ToolSet](https://docs.stackone.com/agents/typescript/frameworks/vercel-ai-sdk)** - Agentic integrations for hundreds of [enterprise SaaS](https://www.stackone.com/integrations) platforms.
 - **[agentic](https://docs.agentic.so/marketplace/ts-sdks/ai-sdk)** - A collection of 20+ tools that connect to external APIs such as [Exa](https://exa.ai/) or [E2B](https://e2b.dev/).
+- **[AWS Bedrock AgentCore built-in tools](https://github.com/aws/bedrock-agentcore-sdk-typescript)** - Fully managed [Code Interpreter and Browser](https://docs.aws.amazon.com/bedrock-agentcore/latest/devguide/built-in-tools.html) tools in isolated environments with zero infrastructure management.
```
Can we say "Amazon Bedrock AgentCore Browser & Code Interpreter" we are planning to change that docs URL very soon and the PMM team has guided to call them with their name not just tools as people don't know what they are exactly if we do so.

kosti-v commented on content/tools-registry/registry.ts:
```diff
@@ -147,4 +147,61 @@ console.log(text);`,
   },
+  {
+    slug: 'bedrock-agentcore',
+    name: 'AWS Bedrock AgentCore',
```
Amazon Bedrock AgentCore

Commit:
docs: highlight Browser and Code Interpreter by name in AgentCore descriptions

Update AWS Bedrock AgentCore documentation to explicitly call out Browser
and Code Interpreter as distinct services rather than generic "built-in tools".
This improves product awareness and helps developers understand the specific
capabilities of each service.

Changes:
- Call out Browser and Code Interpreter by name with bold formatting
- Add detailed descriptions from AgentCore documentation for each service
- Browser: fast, secure cloud-based runtime for web interaction
- Code Interpreter: isolated sandbox for Python/JS/TS code execution

These changes address PM feedback to improve service discoverability and
better communicate the value of each AgentCore component.

nicoalbanese committed:
update tags

siwachabhi committed:
Merge branch 'main' into feat/add-bedrock-agentcore-tools
